### PR TITLE
Table onRow and selectedRowKey

### PR DESCRIPTION
**Pull request by rtkaaho**
onRow: Supports onClick events for now
selectedRowKey: can be used to determine which row should be highlighted

### CODE REVIEW

File: src/components/table/Body.tsx [hunk 1]
```diff
@@ -37,11 +62,22 @@ const EmptyContentContainer = styled.div`
...
+export const Body = <T extends { key: string | number }>(
+  props: BodyProps<T>
+) => {
+  const { columns, data, emptyComponent, onRow, selectedRowKey } = props;
 
   const theme = useTheme();
 
+  const handleClick = React.useCallback(
+    (e, d) => {
+      if (onRow && onRow.onClick) {
```
Comment (rtktpeng) [inline src/components/table/Body.tsx:74]: `(e, d) => onRow?.onClick?.(e, d)` 😝

File: src/components/table/Table.tsx [hunk 2]
```diff
@@ -13,6 +13,10 @@ import { TableLoading } from './TableLoading';
 
 export type Justify = 'flex-start' | 'center' | 'flex-end';
 
+export interface OnRowProps<T> {
+  onClick: (e: React.SyntheticEvent<HTMLElement>, record: T) => void;
```
Comment (rtktpeng) [inline src/components/table/Table.tsx:17]: I like this, looks very extendable 👍

File: src/components/table/Table.tsx [hunk 3]
```diff
@@ -34,9 +38,12 @@ export interface ColumnProps<T> {
 
   /** if true, the column will be sortable and be provided in the `onSort` callback when clicked */
   sortable?: boolean;
+
+  /** if true, the column will be visible */
+  visible?: boolean;
```
Comment (rtktpeng) [inline src/components/table/Table.tsx:43]: Is this implemented?

**Review by rtktpeng — APPROVED** [review]

File: src/components/table/Body.tsx (hunk 1 above)
Comment (rtkaaho) [reply src/components/table/Body.tsx:74]: For some reason storybook didnt like that syntax 🤷‍♂️ . I assume I need to update something in the config. Something for a later date.

File: src/components/table/Table.tsx (hunk 3 above)
Comment (rtkaaho) [reply src/components/table/Table.tsx:43]: oops no its not. I will leave it and implement it shortly. victor made a ticket about it .

**Pull request merged**